File: Python/Algorithms/404.py
class Solution:
    def sumOfLeftLeaves(self, root) -> int:
        def yield_left_values(node):
            left, right = node.left, node.right

            # if left has no children, it is a leaf,
            # yield its value.
            if left:
                if not left.left and not left.right:
                    yield left.val
                else:
                    yield from yield_left_values(left)
            if right:
                yield from yield_left_values(right)

        # pass generator to sum and return.
        return sum(yield_left_values(root), 0)

File: Javascript/Algorithms/404.js
let sumOfLeftLeaves = root => {
  let sum = 0;
  // iife. no need to keep it around.
  (function sum_left_leaves(node) {
    const left = node.left, right = node.right;

    if (left) {
      if (!left.left && !left.right)
        sum += left.val;
      else
        sum_left_leaves(left);
    }
    if (right) sum_left_leaves(right);
  })(root)
  return sum;
};